api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -322,6 +327,19 @@ private User initializeUserIfNeeded() {
     return getProfileResponse(user);
   }
 
+  @Override
+  public ResponseEntity<Profile> verifyEmail(VerifyEmailRequest request) {
+    User user = userDao.findUserByEmail(request.getUsername());
+    try {
+      mailChimpService.addUserContactEmail(user.getContactEmail());
```

Comment:
Seems like we might want to record the fact that we requested email verification (and the time it happened) here.

That way we can display in the UI "check your email" or whatever.

Reply:
The function returns the updated profile which triggers the UI (if it successfully submitted to MailChimp) to display a message (see https://github.com/all-of-us/workbench/pull/460/files/49fadbd89d5f1a5a0a22b2ea5a2c4655b0f16166#diff-ba01bcfd07f9b284c43d41c583d56a83R32).

Reply:
Thought on how we do this and the caching below: add email verification status enum as a field on User. Starts out as UNVERIFIED, goes to PENDING when this method is called successfully; we keep checking mailchimp while status is PENDING, and then once its verified we switch it to MEMBER.

Reply:
@rsasch I'm suggesting that we cache the status of email verification on User rather than hitting mailchimp on every call. 

I guess if User is in a pending state we'll have to hit mailchimp anyway to see if the email has been verified, so that doesn't save us much... but once you're verified it'd be nice to never have to check it again.

Reply:
Agreed that we should probably cache the result. We definitely need to keep pinging mailchimp if status is pending though.